apps/system/js/base_module.js

```diff
@@ -110,7 +102,83 @@
    */
   BaseModule.STATES = [];
 
+  BaseModule.SIDE_MODULES = [];
```

Comment:
Not sure about the "side modules" name, but not sure I fully understand the concept either.
Are those "lazy modules"? If yes I think `LAZY_MODULES` and `LayModuleMixin` would be more explicit (and a comment block would probably help too).